fedimint-server/src/config.rs

```diff
@@ -64,52 +77,152 @@ pub struct Peer {
     pub name: String,
 }
 
+pub struct FederationeConfigGenParams {
+    pub mint_amounts: Vec<Amount>,
+    pub bitcoin_rpc: BitcoindRpcCfg,
+
+    /// extra options for extra settings and modules
+    pub other: BTreeMap<String, serde_json::Value>,
+}
+
 #[derive(Debug, Clone)]
 /// network config for a server
 pub struct ServerConfigParams {
     pub tls: TlsConfig,
     pub hbbft: NetworkConfig,
     pub api: NetworkConfig,
     pub server_dkg: NetworkConfig,
+    // TODO: delete unused dkgs
     pub wallet_dkg: NetworkConfig,
     pub lightning_dkg: NetworkConfig,
     pub mint_dkg: NetworkConfig,
     pub amount_tiers: Vec<Amount>,
     pub federation_name: String,
     pub bitcoind_rpc: String,
+
+    /// extra options for extra settings and modules
+    pub other: BTreeMap<String, serde_json::Value>,
 }
 
-#[async_trait(?Send)]
-impl GenerateConfig for ServerConfig {
-    type Params = HashMap<PeerId, ServerConfigParams>;
-    type ClientConfig = ClientConfig;
-    type ConfigMessage = (KeyType, DkgMessage<G1Projective>);
-    type ConfigError = ();
+impl ServerConfig {
+    pub fn to_client_config(&self) -> ClientConfig {
+        let nodes = self
+            .peers
+            .iter()
+            .map(|(peer_id, peer)| Node {
+                url: peer.api_addr.clone(),
+                name: format!("node #{}", peer_id),
```

Comment:
I thought we provided a way for guardians to pass in a custom node name.

Reply:
I don't recall writing this part myself so I must have copied it from somewhere.